### ISSUE TITLE
Wifi Device Configuration does not save Settings via LUCI web Interface

### ISSUE THREAD

**Issue by MichaelUray** [openwrt/openwrt#17337]
### Describe the bug

**The operating frequency mode is empty per default:**

![2024-12-22_12-40-45_AP252_-_LuCI_–_Mozilla_Firefox](https://github.com/user-attachments/assets/77c69e2f-364b-4729-a335-355c0c395012)

**If I try to change any settings, it only keeps rotating the circle in the save button, but it does not change anything in the configuration.**
![2024-12-22_12-41-17_AP252_-_LuCI_–_Mozilla_Firefox](https://github.com/user-attachments/assets/c49b7c51-e824-4796-a6c6-f0e435094bdb)

### OpenWrt version

r28202-8667ca841b

### OpenWrt release

24.10.0-rc3

### OpenWrt target/subtarget

ipq40xx/mikrotik

### Device

MikroTik hAP ac3

### Image kind

Official downloaded image

### Steps to reproduce

Open the Wifi configuration after a factory reset and try to change any settings there, it will stuck after pushing the save button.

### Other information:

```
root@AP252:~# cat /etc/config/wireless

config wifi-device 'radio0'
        option type 'mac80211'
        option path 'platform/soc/a000000.wifi'
        option band '2g'
        option channel '1'
        option htmode 'VHT20'

config wifi-iface 'default_radio0'
        option device 'radio0'
        option network 'lan'
        option mode 'ap'
        option ssid 'OpenWrt'
        option encryption 'none'

config wifi-device 'radio1'
        option type 'mac80211'
        option path 'platform/soc/a800000.wifi'
        option band '5g'
        option channel '36'
        option htmode 'VHT80'
        option disabled '1'

config wifi-iface 'default_radio1'
        option device 'radio1'
        option network 'lan'
        option mode 'ap'
        option ssid 'OpenWrt'
        option encryption 'none'

```

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by samara1532** [comment]
MT7921 - rockchip/armv8 - 24.10.0.rc3

Wifi Device Configuration does not save Settings

**Comment by es86de** [comment]
Same problem with saving on TP-Link WDR3600 running 24.10.0-rc3.

**Comment by MichaelUray** [comment]
I just tested it with `OpenWrt 24.10.0-rc2 r28161-ea17e958b9`:
The mode is also not set to `N` per default, but I can change settings and save/apply them.

**Comment by namiltd** [comment]
Similar issue with configuration not being saved: https://github.com/openwrt/openwrt/issues/17333

**Comment by namiltd** [comment]
Fixed in https://github.com/openwrt/luci/commit/8a422af6a0e97cdb7b3df1049898731a4b2d153d
The change will be included in the next rc version.

**Comment by MichaelUray** [comment]
> Fixed in [openwrt/luci@8a422af](https://github.com/openwrt/luci/commit/8a422af6a0e97cdb7b3df1049898731a4b2d153d)

Thanks, this could get used as workaround for the moment in -rc3:
```
sed -i 's/if(c\[conf\]?\.\[sid\]\[opt\]!=null)/if(c\[conf\]?\.\[sid\]?\.\[opt\]!=null)/g' /www/luci-static/resources/uci.js
```

You might have to clear the browser cache afterwards or open it in an anonymous window.

**Comment by namiltd** [comment]
OpenWrt 24.10-rc4 is ready (https://github.com/openwrt/openwrt/commit/1e530e5831f3a0e10576d106236f1fa6f48c39e4). Now all you have to do is wait for the images to be generated.

**Comment by hauke** [comment]
The problem only appeared in 24.10.0-rc3. It should be fixed in 24.10.0-rc4.
